### PR TITLE
ENG-128 added support for configuring cors headers from systemParams

### PR DESCRIPTION
**Pull request by ffleandro**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/com/agiletec/aps/system/services/baseconfig/BaseConfigManager.java [hunk 1]
```diff
@@ -119,6 +120,18 @@ public String getParam(String name) {
...
     }
 
+    /**
+     * Retrieves property from Environment variable. If not present fallbacks to {@code systemParams}
+     * Example: {@code MY_SYSTEM_ENV_VARIABLE} fallbacks to {@code systemParams} with key {@code my.system.env.variable}
+     * @param name Parameter key
+     * @return Parameter value
+     */
+    @Override
+    public String getProperty(String name) {
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/baseconfig/BaseConfigManager.java:130]: Spring does this for us for free. This isn't necessary

File: engine/src/main/java/com/agiletec/aps/system/services/baseconfig/ConfigInterface.java [hunk 2]
```diff
@@ -53,14 +60,38 @@
...
+    /**
+     * Retrieves property from Environment variable.
+     * If not present fallbacks to manually provided {@code Property} property map.
+     * Example: {@code MY_SYSTEM_ENV_VARIABLE} fallbacks to properties map with key {@code my.system.env.variable}
+     *
+     * @param properties Fallback property map
+     * @param name Parameter key
+     * @return Parameter value
+     */
+    public static String getProperty(Properties properties, String name) {
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/baseconfig/ConfigInterface.java:90]: Spring does this for free. We don't need this

File: engine/src/main/java/org/entando/entando/aps/servlet/security/CORSFilter.java [hunk 3]
```diff
@@ -29,17 +32,52 @@
 public class CORSFilter extends OncePerRequestFilter {
 
     private final Logger logger = LoggerFactory.getLogger(getClass());
-    
-    public static final String ALLOWED_METHODS = "GET, POST, PUT, DELETE, OPTIONS, PATCH";
+
+    private String allowedOrigins;
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/servlet/security/CORSFilter.java:36]: Inject these with annotations.
@Value(${CORS_ALLOWED_ORIGIN})

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**